Also, something wrong on Windows:

```
FAIL: test_blob_set_item_negative_index (test.test_sqlite3.test_dbapi.BlobTests.test_blob_set_item_negative_index)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "D:\a\cpython\cpython\Lib\test\test_sqlite3\test_dbapi.py", line 1210, in test_blob_set_item_negative_index
    self.assertEqual(self.blob[-1], 255)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: 36281599 != 255
```

--- a/Modules/_sqlite/blob.c
+++ b/Modules/_sqlite/blob.c
@@ -139,7 +139,7 @@ static PyObject *
 read_multiple(pysqlite_Blob *self, Py_ssize_t length, Py_ssize_t offset)
 {
     assert(length <= sqlite3_blob_bytes(self->blob));
-    assert(offset <= sqlite3_blob_bytes(self->blob));
+    assert(offset < sqlite3_blob_bytes(self->blob));
 
     PyObject *buffer = PyBytes_FromStringAndSize(NULL, length);
     if (buffer == NULL) {
@@ -189,6 +189,11 @@ blob_read_impl(pysqlite_Blob *self, int length)
     int max_read_len = blob_len - self->offset;
     if (length < 0 || length > max_read_len) {
         length = max_read_len;
+    }
+
+    assert(length >= 0);
+    if (length == 0) {
+        return PyBytes_FromStringAndSize(NULL, 0);
     }
 
     PyObject *buffer = read_multiple(self, length, self->offset);

--- a/Modules/_sqlite/blob.c
+++ b/Modules/_sqlite/blob.c
@@ -122,8 +122,6 @@ blob_seterror(pysqlite_Blob *self, int rc)
 static PyObject *
 read_single(pysqlite_Blob *self, Py_ssize_t offset)
 {
-    assert(offset <= sqlite3_blob_bytes(self->blob));
-
     unsigned long buf = 0;
     int rc;
     Py_BEGIN_ALLOW_THREADS

--- a/Modules/_sqlite/blob.c
+++ b/Modules/_sqlite/blob.c
@@ -124,7 +124,7 @@ read_single(pysqlite_Blob *self, Py_ssize_t offset)
 {
     assert(offset <= sqlite3_blob_bytes(self->blob));
 
-    unsigned long buf;
+    unsigned long buf = 0;
     int rc;
     Py_BEGIN_ALLOW_THREADS
     rc = sqlite3_blob_read(self->blob, (void *)&buf, 1, (int)offset);

